### ISSUE TITLE
sqlfluff-fix pre-commit hook breaks INSERT statements due to column reordering

### ISSUE THREAD

**Issue by armaandhull** [sqlfluff/sqlfluff#1268]
This issue came up when using the pre-commit hook `sqlfluff-fix` as mentioned in the documentation https://docs.sqlfluff.com/en/stable/production.html#using-pre-commit

## Expected Behaviour
After running `sqlfluff-fix`, the output should be
```
INSERT INTO example_schema.example_table
(id,
    example_column,
    rank_asc,
    rank_desc)
SELECT
    id,
    CASE
        WHEN col_a IN('a',
                              'b',
                'c') THEN col_a
    END AS example_column,
    rank_asc,
    rank_desc
FROM
  another_schema.another_table
```

## Observed Behaviour
Running the pre-commit hook `sqlfluff-fix` reorders columns in the SELECT clause without making the same modification in the INSERT clause, thus causing an error.
```
INSERT INTO example_schema.example_table
(id,
    example_column,
    rank_asc,
    rank_desc)
SELECT
    id,
    rank_asc,
    rank_desc,
    CASE
        WHEN col_a IN('a',
                              'b',
                'c') THEN col_a
    END AS example_column
FROM
  another_schema.another_table
```
## Steps to Reproduce
- add the example SQL to a file
- `git add` the file
- `git commit -m "testing my change"`
- check the fixed file after `sqlfluff-fix` pre-commit hook has finished fixing the SQL

## Version
```
sqlfluff, version 0.6.2
Python 3.7.6
```
## Configuration
```
[sqlfluff]
verbose=1
dialect=snowflake
ignore=parsing,lexing
max_line_length=200

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy=upper

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy=lower

[sqlfluff:rules:L030]  # Function names
capitalisation_policy=lower

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy=upper
```

**Comment by barrywhart** [comment]
Yow, that's bad! I think this is happening due to rule [L034](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L034). Can you try disabling that rule using `exclude_rules` to see if that prevents the problem (until we get this fixed)?

You can see an example of disabling a rule [here](https://stackoverflow.com/questions/66878455/configure-sqlfluff-rules).

**Comment by armaandhull** [comment]
@barrywhart yes, I was able to narrow it down to rule `L034` and exclude it in our pre-commit config.

**Comment by barrywhart** [comment]
To fix this, it may be simpler for the rule to detect if it's running "inside" an `INSERT` statement and simply do nothing (rather than trying to update both the `INSERT` and the `SELECT`).

**Comment by armaandhull** [comment]
@barrywhart that makes sense. The 3 variations that I can think of are:

- running "inside" an INSERT statement
- running "inside" a CREATE TABLE AS statement
- running "inside" a COPY INTO <table> statement (used by Snowflake)

**Comment by barrywhart** [comment]
@armaandhull: I have a PR (#1384) which addresses the `INSERT` and `CREATE TABLE AS SELECT` cases. As far as I can tell, the Snowflake `COPY INTO` syntax is not yet supported by SQLFluff. Can you create a separate issue about adding support for that?

There'll need to be another change to L034 to handle `COPY INTO` once that work is done.

**Comment by armaandhull** [comment]
thanks @barrywhart, created a separated issue for `COPY INTO` https://github.com/sqlfluff/sqlfluff/issues/1397

**Comment by tunetheweb** [comment]
FYI @barrywhart COPY INTO was added to Snowflake in #1353 (which landed in 0.6.5) so I presume your fix (which was #1383 so after that) already included the necessary changes for that?

**Comment by barrywhart** [comment]
I don't think it did. I'll double check.

**Comment by barrywhart** [comment]
From the discussion above, my PR did not cover `COPY INTO`. Note my PR (#1384) was merged 2 days prior to closing the issue about adding `COPY INTO` support (#1397).

I wonder if there was partial `COPY INTO` support prior to that, but not the "flavor" that involved a list of columns? IIRC, I took a quick look at the Snowflake dialect, tried an example of the syntax as I was able to get from the docs, and it didn't work, so I skipped it. I was out of scope wrt the original issue, which mentioned `INSERT`, and I wanted to avoid delaying a fix due to scope creep, especially delaying an ANSI fix for something specific to one dialect.
